src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -472,9 +472,28 @@ def delete_lock(name, resource_group_name=None, resource_provider_namespace=None
 
 def create_lock(name, resource_group_name=None, resource_provider_namespace=None,
                 parent_resource_path=None, resource_type=None, resource_name=None,
-                level=None, notes=None, lock_id=None, lock_type=None):
+                level=None, notes=None, lock_id=None):
+    '''
+    :param name: The name of the lock.
+    :type name: str
+    :param resource_provider_namespace: Name of a resource provider.
+    :type resource_provider_namespace: str
+    :param parent_resource_path: Path to a parent resource
+    :type parent_resource_path: str
+    :param resource_type: The type for the resource with the lock.
+    :type resource_type: str
+    :param resource_name: Name of a resource that has a lock.
+    :type resource_name: str
+    :param level: The locking level, must be either 'ReadOnly' or 'CanNotDelete'
+    :type level: str
```

Comment:
> Could this enum be specified in a way that would support tab completion?

Agreed. The param definitions here should go into _params.py. In there, you can specify help for each param and the list of possible values (then tab completion would work with that).